contrib/golang/filters/http/source/go/pkg/http/type.go

```diff
@@ -32,32 +32,45 @@ const (
 	errInvalidPhase    = "invalid phase, maybe headers/buffer already continued"
 )
 
-type httpHeaderMap struct {
+// api.HeaderMap
+type headerMapImpl struct {
 	request     *httpRequest
 	headers     map[string][]string
 	headerNum   uint64
 	headerBytes uint64
-	isTrailer   bool
 }
 
-var _ api.HeaderMap = (*httpHeaderMap)(nil)
-
-func (h *httpHeaderMap) GetRaw(key string) string {
-	if h.isTrailer {
-		panic("unsupported yet")
+func (h *headerMapImpl) Range(f func(key, value string) bool) {
+	if h.headers == nil {
+		h.headers = cAPI.HttpCopyHeaders(unsafe.Pointer(h.request.req), h.headerNum, h.headerBytes)
```

Comment:
this is only for headers, should move to `requestOrResponseHeaderMapImpl`.

Reply:
done.